src/m365/teams/commands/user/user-app-list.ts

```diff
@@ -0,0 +1,114 @@
+import * as chalk from 'chalk';
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import Utils from '../../../../Utils';
+import { GraphItemsListCommand } from '../../../base/GraphItemsListCommand';
+import request from '../../../../request';
+import commands from '../../commands';
+import { UserTeamsApp } from '../../UserTeamsApp';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  userId: string;
+  userName: string;
+}
+
+class TeamsUserAppListCommand extends GraphItemsListCommand<UserTeamsApp> {
+  public get name(): string {
+    return `${commands.TEAMS_USER_APP_LIST}`;
+  }
+
+  public get description(): string {
+    return 'Lists the apps deployed in the personal scope of the specified user';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    var userIdPromise: Promise<{ value: string; }>;
+
+    if(args.options.userName) {
+      const requestOptions: any = {
+        url: `${this.resource}/v1.0/users/${encodeURIComponent(args.options.userName)}/id`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
+
+      userIdPromise = request.get<{ value: string; }>(requestOptions);  
+    } else {
+      userIdPromise = Promise.resolve({ value : args.options.userId });
+    }
+
+    userIdPromise.then((userId) => {
+      const endpoint: string = `${this.resource}/beta/users/${encodeURIComponent(userId.value)}/teamwork/installedApps`
+      
+      this.getAllItems(endpoint, logger, true)
```

Comment:
We should avoid nesting promises and instead return this operation to the main chain.